src/main/java/maria/task/TaskList.java

```diff
@@ -0,0 +1,80 @@
+package maria.task;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import maria.Storage;
+import maria.util.StorageConverter;
+
+public class TaskList extends ArrayList<Task> {
+
+    private final Storage storage;
+
+    public TaskList(Storage storage) {
+        super();
+        this.storage = storage;
+    }
+
+    private void syncStorageState() {
+        try {
+            this.storage.writeToFile(StorageConverter.tasksToString(this));
+        } catch (IOException e) {
+            System.out.println("Sync storage state failed. " + e.getMessage());
+        }
+    }
+
+    /**
+     * Adds a task as per the standard library List, and syncs with the storage file.
+     * @param task The task to be added
+     * @return Always true
+     */
+    @Override
+    public boolean add(Task task) {
+        boolean res = super.add(task);
+        syncStorageState();
+        return res;
+    }
+
+    /**
+     * Removes a task as per the standard library List, and syncs with the storage file.
```

Comment:
Good job with the JavaDoc descriptions! Almost all of them are clear and descriptive